astropy/stats/biweight.py

```diff
@@ -397,9 +402,15 @@ def biweight_midvariance(data, c=9.0, M=None, axis=None,
 
     if axis is None:
         if mad == 0.:  # data is constant or mostly constant
-            return 0.0
```

Comment:
@larrybradley - rather than the `Quantity` check, I *think* that if you just `return mad` here, the units will already be in place!

Reply:
@mhvk I like that (but need to return `mad ** 2` for variance units).  The problem I ran into is that `median_absolute_deviation` also drops units if the input contains NaN.  I tracked that back to a bug(?) where `np.median` *also* drops units for NaN:

```python
>>> import numpy as np
>>> import astropy.units as u
>>> data = np.array([1., 2, np.nan, 3, 4]) << u.km
>>> np.median(data)
nan
>>> np.mean(data)
<Quantity nan km>
>>> np.var(data)
<Quantity nan km2>
```

I added a workaround in `median_absolute_deviation` to fix the `np.median` issue.  It feels a bit hacky, but I don't have a better solution.

Reply:
Yikes, definitely a bug! I copied your example to https://github.com/astropy/astropy/issues/12165

Reply:
This part now looks great!